src/qengine/opencl.cpp

```diff
@@ -1637,8 +1589,15 @@ void QEngineOCL::INTBCD(OCLAPI api_call, bitCapInt toMod, const bitLenInt start,
     if (nibbleCount * 4 != length) {
         throw std::invalid_argument("BCD word bit length must be a multiple of 4.");
     }
-    bitCapInt inOutMask = ((1 << length) - 1) << start;
-    bitCapInt otherMask = ((1 << qubitCount) - 1) ^ inOutMask;
+
+    bitCapInt maxPow = intPow(10U, nibbleCount);
+    toMod %= maxPow;
+    if ((length == 0U) || (toMod == 0U)) {
+        return;
+    }
+
+    bitCapInt inOutMask = ((1U << length) - 1U) << start;
```

Comment:
This is a super common convention.  Push it into an inline function in the base class so that everyone can call "inOutMask = makeBitCapMask(length, start)"

Ditto for the other standard mask functions.  If they're super common, consider making a macro.

Reply:
Done.

In the case you specifically pointed to, we don't need both a "length mask" and a "register mask, just the register mask. I made an inline function, for this common case. When we need both, the function or macro is literally a single bit shift on 1U, so I think leaving that out is a happy middle.